molgenis-security-core/src/main/java/org/molgenis/security/core/utils/SecurityUtils.java

```diff
@@ -28,46 +27,55 @@ private SecurityUtils() {}
   /**
    * Returns the username of the current authentication.
    *
-   * @return username or <tt>null</tt> if 1) the current authentication is null or 2) the currently
-   *     authenticated principal is the system.
+   * @return a username, null if the current authentication is null
```

Comment:
Sorry, I think I found one case where this breaks: HttpLocaleResolver.resolveLocale happily falls back to the system locale when the user name is null but gets annoyed when the user name cannot be found in the database.

Reply:
Update: but I cannot trigger it because exceptions thrown by a run as system block get translated outside the block